sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -463,6 +463,32 @@ public String getId() {
         return queryItemsInternal(querySpec, options, classType);
     }
 
+    /**
+     * Query for items in the current container using a {@link SqlQuerySpec} and {@link CosmosQueryRequestOptions}.
+     * <p>
+     * After subscription the operation will be performed. The {@link Flux} will
+     * contain one or several feed response of the obtained items. In case of
+     * failure the {@link CosmosPagedFlux} will error.
+     *
+     * @param <T> the type parameter.
+     * @param querySpec the SQL query specification.
+     * @param options the query request options.
+     * @param classType the class type.
+     * @param feedRange the feedrange to query
+     * @return a {@link CosmosPagedFlux} containing one or several feed response pages of the obtained items or an
+     * error.
+     */
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public <T> CosmosPagedFlux<T> queryItems(SqlQuerySpec querySpec, CosmosQueryRequestOptions options,
+                                             Class<T> classType, FeedRange feedRange) {
```

Comment:
`CosmosQueryRequestOption` has `CosmosQueryRequestOption:setPartition(.)`
FeedRange to me seems a very similar concept to PartitionKey.

Maybe we should add `CosmosQueryRequestOption.setFeedRange`? and drop feedRange as a 4th parameter in this method?

Reply:
PartitionKey and FeedRange overlap on concepts. In .net we opted for only FeedRange, so that they don't get confused. 

---
In reply to: [577295115](https://github.com/Azure/azure-sdk-for-java/pull/19275#discussion_r577295115) [](ancestors = 577295115)

Reply:
we can leave this discussion for later as this is Beta API, we can change it anytime later.

IMO, `FeedRange` should be part of the `CosmosQueryRequestOptions`. a method call with 4 args is very confusing and overkill.

Reply:
+1 here. And I would actually make that change within the PR - at least if my assumption that this would only take a coule of hours is correct.

Reply:
+1 on moving this to query options which would also be more consistent with change feed pull model. I would prefer making this change within the scope of this PR.